File: bubble_sort/java/BubbleSort.java
public class BubbleSort {
    public static void bubbleSort(int[] array) {
        int n = array.length;
        boolean swapped;
        do {
            swapped = false;
            for (int i = 1; i < n; i++) {
                if (array[i - 1] > array[i]) {
                    // swap array[i - 1] and array[i]
                    int temp = array[i - 1];
                    array[i - 1] = array[i];
                    array[i] = temp;
                    swapped = true;
                }
            }
            n--; // Decrease n because the last element is now in its correct place
        } while (swapped);
    }

    public static void main(String[] args) {
        int[] numbers = {64, 34, 25, 12, 22, 11, 90};
        System.out.println("Unsorted array:");
        for (int number : numbers) {
            System.out.print(number + " ");
        }
        System.out.println();

        bubbleSort(numbers);

        System.out.println("Sorted array:");
        for (int number : numbers) {
            System.out.print(number + " ");
        }
    }
}

File: bubble_sort/php/bubble_sort.php
<?php
function bubbleSort($array) {
    $n = count($array);
    for ($i = 0; $i < $n - 1; $i++) {
        for ($j = 0; $j < $n - $i - 1; $j++) {
            if ($array[$j] > $array[$j + 1]) {
                // Swap $array[$j] and $array[$j + 1]
                $temp = $array[$j];
                $array[$j] = $array[$j + 1];
                $array[$j + 1] = $temp;
            }
        }
    }
    return $array;
}

// Example usage
$unsortedArray = array(64, 34, 25, 12, 22, 11, 90);
$sortedArray = bubbleSort($unsortedArray);

echo "Sorted array: \n";
print_r($sortedArray);
?>
